core/src/client/mod.rs

```diff
@@ -0,0 +1,60 @@
+use derivative::Derivative;
+
+use crate::{config::Config, metadata::Metadata};
+
+/// the base for a client should be:
+/// - runtime version
+/// - genesis hash
+/// - metadata
+
+#[derive(Derivative)]
+#[derivative(Debug(bound = ""), Clone(bound = ""))]
+pub struct ClientBase<C: Config> {
+    pub genesis_hash: C::Hash,
+    pub runtime_version: RuntimeVersion,
+    pub metadata: Metadata,
+    marker: core::marker::PhantomData<C>,
+}
+
+impl<C: Config> ClientBase<C> {
+    pub fn new(genesis_hash: C::Hash, runtime_version: RuntimeVersion, metadata: Metadata) -> Self {
+        Self {
+            genesis_hash,
+            runtime_version,
+            metadata,
+            marker: core::marker::PhantomData,
+        }
+    }
+
+    pub fn metadata(&self) -> Metadata {
+        self.metadata.clone()
```

Comment:
This is super expensive to clone (unlike the subxt version of the Metadata it's not wrapped in an Arc, so we are cloning the entire metadata structure and everything in it every time).

As such, cloning `ClientBase` is also super expensive (whic his currently necessary because SignedExtensions are all asking for an owned version of it, for instance).

I'd be tempted to remove `Clone` from `Metadata` and `ClientBase` as a test to see how many places we are relying on it, and then see what we can do to pass it by reference instead in this no-std layer (since things shouldn't be needing to mutate the metadata or client base stuff anyways)

Reply:
Ooh, no I was wrong; you are pointing at a metadata which is stilkl wrapped in Arc, so ignore this :)